fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/load/ChunkWriter.java

```diff
@@ -298,24 +309,33 @@ public void writeItems(List<java.lang.Object> arg0) throws Exception {
      * @throws FHIRPersistenceException
      */
     public OperationOutcome conditionalFingerprintUpdate(ImportTransientUserData chunkData, boolean skip, Map<String, SaltHash> localCache, FHIRPersistence persistence, FHIRPersistenceContext context, String logicalId, Resource resource) throws FHIRPersistenceException {
+        
+        // Since issue 1869, we always need to always perform the read here in order to obtain the
+        // latest version id. When we call the persistence layer, the resource should already
+        // be updated with the correct id/meta values
+        Resource oldResource = null;
+        try {
+            // This execution is in a try-catch-block since we want to catch
+            // the resource deleted exception.
+            oldResource = persistence.read(context, resource.getClass(), logicalId).getResource();
+        } catch (FHIRPersistenceResourceDeletedException fpde) {
+            logger.throwing("ChunkWriter", "conditionalFingerprintUpdate", fpde);
+        }
```

Comment:
FHIRRestHelper.doRead takes a boolean flag for controlling whether or not we allow reading of the deleted resource.  Specifically, it gets set in the FHIRPersistenceContext and then used to control whether the PL throws in this case or not:
```
            if (resourceIsDeleted && !context.includeDeleted()) {
                throw new FHIRPersistenceResourceDeletedException("Resource '" +
                        resourceType.getSimpleName() + "/" + logicalId + "' is deleted.");
            }
```

Do you know if we're setting `context.includeDeleted()` one way or the other?
I think we should be setting it to true and therefor we'd never except to hit this particular catch block.
If that is right, we might want to throw (or at least log a warning if we hit this unexpected case).